coherence/v1client.go

```diff
@@ -0,0 +1,1371 @@
+		defer cancel()
+	}
+
+	defer m.cleanupRequest(req.Id)
+
+	result, err1 := waitForResponse(newCtx, requestType.ch)
+	if err1 != nil {
+		return value, err1
+	}
+
+	return unwrapBool(result)
+}
+
+// get issues a get request for a given key and returns the bytes value which could be nil.
+func (m *streamManagerV1) get(ctx context.Context, cache string, key []byte) (*[]byte, error) {
+	req, err := m.newGetRequest(cache, key)
+	if err != nil {
+		return nil, err
+	}
+
+	requestType, err := m.submitRequest(req, pb1.NamedCacheRequestType_Get)
+	if err != nil {
+		return nil, err
+	}
+
+	newCtx, cancel := m.session.ensureContext(ctx)
+	if cancel != nil {
+		defer cancel()
+	}
+
+	// remove the entry from the channel
+	defer m.cleanupRequest(req.Id)
+
+	result, err1 := waitForResponse(newCtx, requestType.ch)
+	if err1 != nil {
+		return nil, err1
+	}
+
+	var message = &pb1.OptionalValue{}
+	if err = result.UnmarshalTo(message); err != nil {
+		err = getUnmarshallError("getResponse", err)
+		return nil, err
+	}
+	// set the return value based upon the optional
+	if message.Present {
+		return &message.Value, nil
+	}
+	// no value present, return nil
+	return nil, nil
+}
+
+// mapListenerRequest issues a map listener request to add or remove a key or filter listener.
+func (m *streamManagerV1) mapListenerRequest(ctx context.Context, cache string, subscribe bool, keyOrFilter *pb1.KeyOrFilter,
+	lite bool, synchronous bool, priming bool, filterID int64) error {
+
+	req, err := m.newMapListenerRequest(cache, subscribe, keyOrFilter, filterID, lite, synchronous, priming, nil)
+	if err != nil {
+		return err
+	}
+
+	requestType, err := m.submitRequest(req, pb1.NamedCacheRequestType_MapListener)
+	if err != nil {
+		return err
+	}
+
+	newCtx, cancel := m.session.ensureContext(ctx)
+	if cancel != nil {
+		defer cancel()
+	}
+
+	// remove the entry from the channel
+	defer m.cleanupRequest(req.Id)
+
+	_, err1 := waitForResponse(newCtx, requestType.ch)
+	if err1 != nil {
+		return err1
+	}
+
+	return nil
+}
+
+func (m *streamManagerV1) replaceMapping(ctx context.Context, cache string, key []byte, prevValue []byte, newValue []byte) (bool, error) {
+	req, err := m.newReplaceMappingRequest(cache, key, prevValue, newValue)
+	if err != nil {
+		return false, err
+	}
+
+	requestType, err := m.submitRequest(req, pb1.NamedCacheRequestType_ReplaceMapping)
+	if err != nil {
+		return false, err
+	}
+
+	newCtx, cancel := m.session.ensureContext(ctx)
+	if cancel != nil {
+		defer cancel()
+	}
+
+	// remove the entry from the channel
+	defer m.cleanupRequest(req.Id)
+
+	result, err1 := waitForResponse(newCtx, requestType.ch)
+	if err1 != nil {
+		return false, err1
+	}
+
+	return unwrapBool(result)
+}
+
+// remove issues a get request for a given key and returns the bytes value which could be nil.
+func (m *streamManagerV1) remove(ctx context.Context, cache string, key []byte) (*[]byte, error) {
+	req, err := m.newRemoveRequest(cache, key)
+	if err != nil {
```

Comment:
The code from 601 to 621 seems to be repeated quite a bit.  Could this be abstracted?

Reply:
good catch have made generic function to replace this